doc/ntp/ntp-design.md

```diff
@@ -0,0 +1,898 @@
+            enum yes;
+            enum no;
+        }
+    }
+
+    typedef ntp-key-type {
+        description "NTP key encrypt type";
+        type enumeration {
+            enum md5;
+            enum sda1;
+        }
+    }
+
+    typedef association-type {
+        description "NTP server association type";
+        type enumeration {
+            enum server;
+            enum peer;
+            enum pool;
+        }
+    }
+
+    container sonic-ntp {
+
+        container NTP {
+
+            container global {
+
+                description "Global NTP part of config_db.json";
+
+                leaf-list src_intf {
+                    type union {
+                        type leafref {
+                            path /port:sonic-port/port:PORT/port:PORT_LIST/port:name;
+                        }
+                        type leafref {
+                            path /lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:name;
+                        }
+                        type leafref {
+                            path /lo:sonic-loopback-interface/lo:LOOPBACK_INTERFACE/lo:LOOPBACK_INTERFACE_LIST/lo:name;
+                        }
+                        type leafref {
+                            path /mprt:sonic-mgmt_port/mprt:MGMT_PORT/mprt:MGMT_PORT_LIST/mprt:name;
+                        }
+                    }
+
+                }
+
+                leaf vrf {
+                    must "(current() != 'mgmt') or (/mvrf:sonic-mgmt_vrf/mvrf:MGMT_VRF_CONFIG/mvrf:vrf_global/mvrf:mgmtVrfEnabled = 'true')" {
+                        error-message "Must condition not satisfied. Try enable Management VRF.";
+                    }
+
+                    type string {
+                        pattern "mgmt|default";
+                    }
+
+                    description
+                        "NTP can be enabled only in one VRF at time. In this revision, it is either default VRF or Management VRF.";
+                }
+
+                leaf state {
+                    type state;
+                    default enabled;
+                    description "NTP feature state";
+                }
+
+                leaf authentication {
+                    type state;
+                    default disabled;
+                    description "NTP authentication state";
+                }
+
+                leaf server_role {
+                    type state;
+                    default enabled;
+                    description "NTP server functionality state";
+                }
+
+            } /* end of container global */
+
+        } /* end of container NTP */
+
+        container NTP_SERVER {
+
+            description "NTP SERVER part of config_db.json";
+
+            list NTP_SERVER_LIST {
+                max-elements 10;
+
+                key "server_address";
+
+                leaf server_address {
+                    type inet:host;
+                }
+
+                leaf state {
```

Comment:
Same comment as above for the state field.

Reply:
ok